### PR TITLE
Adding CI setup files

### LINKED ISSUE #6: Setup CI
Add a `Jenkinsfile` to setup CI for this repository. CI should run the following Makefile targets in the repository's root folder:
- `check`: to check if language libraries have latest schemas bundled in them
- `test`: to test language libraries' code

### PR DESCRIPTION
**Pull request by ycombinator**
This PR creates the `.ci` folder and adds the necessary files under it to setup Jenkins CI jobs for this repository.

Follow up to #12.
Resolves #6.

### CODE REVIEW

File: .ci/Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    }
+    /**
+     Test the source code.
+     */
+    stage('Test') {
+      steps {
+        cleanup()
+        withMageEnv(){
+          dir("${BASE_DIR}"){
+            sh(label: 'Check',script: 'make test')
```
Comment (cachedout) [inline .ci/Jenkinsfile:57]: What output does `make test` produce? Is it something that needs to be processed with `junit`?

Comment (ycombinator) [reply .ci/Jenkinsfile:57]: At the moment `make test` will simply run `go test -v ./...` under the hood: https://github.com/elastic/package-spec/pull/12/files#diff-d635a9f292641eeeb317ef8c83354e7dR16

Would it be better if it output in a junit-compatible format? We could certainly make that happen with something like https://github.com/jstemmer/go-junit-report.

**Comment by ycombinator** [comment]
Sorry @cachedout, I saw your comments after I saw Marcin's LGTM so I merged the PR already. I will make a follow up PR to address your feedback.

**Comment by cachedout** [comment]
@ycombinator No worries! Thanks.

**Pull request merged**